### DISCUSSION TITLE
Subtree Focus Mode

### DISCUSSION THREAD

**Discussion by eon-s** [godotengine/godot-proposals#4091] [Editor]
What I mean subtree focus mode, well, with 🌟Focus Mode🌟 I don't mean Zen Mode but close (is like zen mode for level designers).

When editing complex scenes in the 2D or 3D viewport, many nodes get in the way, but what if we can just "highlight" the subtree where we want to work and hide/lock the rest?

For example:
I want to edit these weird platforms but enemies, props, tilemap, background, everything, gets in the way.

![imagen](https://user-images.githubusercontent.com/10215987/155252828-d8536bee-7f5d-48ff-a301-959e35a4f627.png)

If I could "focus" on a subtree and temporarly lock the rest of the nodes (and dim with alpha on 2D or render as wireframes or whatever is possible to do on 3D) it will make level design a lot better experience.

![imagen](https://user-images.githubusercontent.com/10215987/155254784-d51d2e79-1942-4a83-bb70-e34d06b19e63.png)

The focused subtree could be highlighted or the rest grayed out in the scene tree too.

**Comment by azur-wolve** [comment]
It reminds me when hiding the selection or all except the selection in Blender

**Reply by eon-s** [reply]
Exactly, is super useful there for modelling and the situation can be similar in complex and cluttered levels/scene assets.

**Reply by yankscally** [reply]
this is awesome. would be a totally welcomed feature for me personally

**Comment by erayzesen** [comment]
It's exactly great idea and I think about this feature when I working with the scene trees. The feature reminded me Adobe Flash , is it right? When we double click on the symbol object on Flash,  we was feeling that we are in the object.

**Reply by eon-s** [reply]
Now that you mention it, yeah, I was thinking more of some prefab edition modes in unity (I don't remember if is still there or was an experiment) and sublevel locking in unreal but more user friendly and within Godot's scene workflows.

**Comment by djrain** [comment]
A subtree sounds like it could be quite limiting - often subtrees contain unrelated or conflicting objects with regards to editing.

For more power and flexibility, I'd suggest more of a filter system, where you can filter by node type, "x" in node name, nodes in group, custom function (defined on the root node, I guess), etc. Ideally it should allow you to save certain groups/filters project-wide. And some kind of "run function x on these nodes" option. Basically a general purpose level-editor helper. Might be big enough to be an optional addon.

Note that doing this in a simple plugin is currently limited by the ability to properly lock nodes from code (at least) #3046

**Reply by azur-wolve** [reply]
it looks that some of that can be done with a tool script
